### PR TITLE
fixed bugs related to edit button in single news page and duplicate of code in sign-in

### PR DESCRIPTION
**Pull request by KimAT25**
#774 [Display news: Single view] 'Edit' button is available for unregistered User;
#776 [Display news: Single view] 'Edit' button is available for registered User in case he/she has not submitted this particular piece of news;
deleted duplicated code in sign-in template;

### CODE REVIEW

File: src/app/component/auth/components/sign-in/sign-in.component.html [hunk 1]
```diff
@@ -25,7 +25,7 @@ <h2>{{'user.auth.sign-in.fill-form' | translate}}</h2>
              name="email"
              [(ngModel)]="userOwnSignIn.email"
              (input)="configDefaultErrorMessage()"
-             [ngClass]="emailControl.invalid && (emailControl.dirty || emailControl.touched ) || backEndError?
+             [ngClass]="emailControl.invalid && (emailControl.dirty || emailControl.touched ) || backEndError || emailErrorMessageBackEnd?
                 'alert-email-validation':'successful-email-validation'"
```
Comment (Gusvarov) [inline src/app/component/auth/components/sign-in/sign-in.component.html:29]: Please, separate logical && and || in different lines for better code readable

File: src/app/component/auth/components/sign-in/sign-in.component.html [hunk 2]
```diff
@@ -46,16 +46,12 @@ <h2>{{'user.auth.sign-in.fill-form' | translate}}</h2>
...
-        {{emailErrorMessageBackEnd}}
-      </div>
       <div class="email-field-margin"
            *ngIf="backEndError">
       </div>
       <label for="password">{{'user.auth.sign-in.password' | translate}}</label>
       <div class="password-input-btn-show-hide"
-           [ngClass]="passwordControl.invalid && (passwordControl.dirty || passwordControl.touched) || backEndError?
+           [ngClass]="passwordControl.invalid && (passwordControl.dirty || passwordControl.touched) || backEndError || passwordErrorMessageBackEnd?
               'alert-password-validation':'successful-password-validation'">
```
Comment (Gusvarov) [inline src/app/component/auth/components/sign-in/sign-in.component.html:55]: Same as line 28

File: src/app/component/auth/components/sign-in/sign-in.component.html (hunk 1 above)
Comment (KimAT25) [reply src/app/component/auth/components/sign-in/sign-in.component.html:29]: Fixed

File: src/app/component/auth/components/sign-in/sign-in.component.html (hunk 2 above)
Comment (KimAT25) [reply src/app/component/auth/components/sign-in/sign-in.component.html:55]: Fixed

File: src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { EcoNewsService } from '../../services/eco-news.service';
 import { EcoNewsModel } from '../../models/eco-news-model';
 import { Subscription } from 'rxjs';
 import { singleNewsImages } from '../../../../../assets/img/icon/econews/single-news-images';
+import { LocalStorageService } from '../../../../service/localstorage/local-storage.service';
```
Comment (mplesha) [inline src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts:7]: You can use alias here as well in other imports and write something like:
import { LocalStorageService } from '@service_path/localstorage/local-storage.service';

File: src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts [hunk 4]
```diff
@@ -17,15 +18,22 @@ export class EcoNewsDetailComponent implements OnInit, OnDestroy {
   private newsItem: EcoNewsModel;
   private newsImage: string;
   private images = singleNewsImages;
+  private userId: number;
```
Comment (mplesha) [inline src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts:21]: You have those variables in template, so - they are data bound properties!
**Data bound properties must also be public** look here - https://angular.io/guide/aot-compiler

File: src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts (hunk 3 above)
Comment (KimAT25) [reply src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts:7]: Fixed

File: src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts (hunk 4 above)
Comment (KimAT25) [reply src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.ts:21]: Fixed

**Pull request merged**